### ISSUE TITLE
Changing league_entries

### ISSUE THREAD

**Issue by renja-g** [renja-g/Rank-Crawler#1]
The problem is that when running main.py
The logger gives the following output:
```
2023-08-10 22:11:29,760 - INFO - Summoner G5 Easy has not changed
2023-08-10 22:11:29,846 - INFO - Summoner WeingottBachus has not changed
2023-08-10 22:11:29,979 - INFO - league_entries has changed from [{...}] to [{...}]
```
(swap of `RANKED_SOLO_5x5` and `RANKED_TFT_DOUBLE_UP`)
but in the database nothing will get changed at any point

---

Recreation:
1. Fill the `.env`
2. Install `requirements.txt`
```
python -m venv .venv

.\.venv\scripts\activate # windows
. .\.venv\bin\activate # mac

pip install -r requirements.txt
```
4. add few summoners (I did 8)
```
python ./add_summoner.py -name "G5 Easy" -region "euw1"
python ./add_summoner.py -name "WeingottBachus" -region "euw1"
python ./add_summoner.py -name "kiting femboys" -region "euw1"
python ./add_summoner.py -name "PrinzessinBubbIe" -region "euw1"
python ./add_summoner.py -name "Zukyami" -region "euw1"
python ./add_summoner.py -name "qyiugweqyeh" -region "euw1"
python ./add_summoner.py -name "tiger woods 1" -region "euw1"
```
5. run `main.py` (maybe a few times if it does not occurs the first update)
```
python ./main.py
```

**Comment by shahuronghe** [comment]
Hey, I tried to run the program, but couldn't verify the test.

**Comment by shahuronghe** [comment]
File "C:\Programs\Python\Python39\lib\site-packages\riotwatcher\LolWatcher.py", line 73, in __init__
    raise ValueError("Either api_key or kernel_url must be set!")
ValueError: Either api_key or kernel_url must be set!

**Comment by renja-g** [comment]
@shahuronghe Thanks for trying, the error looks like you did not set the Riot API Key, is that possible?

You will have to copy the.env.template and rename the copy to just .env and then fill in the right values

**Comment by shahuronghe** [comment]
How to get Riot API key? I am not familiar with Riot

**Comment by renja-g** [comment]
You can use a free development key for testing, you will have to refresh it every 24h, you can get the key from https://developer.riotgames.com

**Comment by shahuronghe** [comment]
Do I need to create the MongoDB account too?

**Comment by renja-g** [comment]
Well, yea 😅

**Comment by renja-g** [comment]
Inside MongoDB, you will have to create a free cluster,
Create a database inside the cluster called "rank-crawler" and create a collection inside rank-crawler called "summoners"

**Comment by shahuronghe** [comment]
would you like to explain the issue over google meet or discord?

**Comment by renja-g** [comment]
Sure, however I'm at school right now and will only be home in like 7h

**Comment by renja-g** [comment]
But did you managed, to set everything up?

**Comment by shahuronghe** [comment]
yes, everything has been set up, want to understand the issue.

**Comment by renja-g** [comment]
did you take a look at the logs?

**Comment by shahuronghe** [comment]
Yeahhh, i have it. What should be logs in the correct scenario?
Whats the usecase?

**Comment by renja-g** [comment]
When you execute `main.py` (after you added a few summoners)
it would be alright if there are changes on the first time, but if you run `main.py` again and the logs still show some changes, something is off

**Comment by renja-g** [comment]
`2023-08-11 07:36:41,742 - INFO - league_entries has changed from ... to ...`

**Comment by shahuronghe** [comment]
I'll check this tomorrow!

**Comment by NotNullBool** [comment]
When I was looking at the code I noticed you get league entries from separate data than your summoner data. You then add league entry to summoner data. You then set your your original summoner to a copy. So summoner data would have a league entry while summoner copy shouldn't.

**Comment by NotNullBool** [comment]
However I could be wrong it was just my understanding from what I've read and the api. This is the first issue I've ever tried working on.

**Comment by renja-g** [comment]
Yes and no, I get the data from two endpoints, but the summoner in the DB also has the `league_entry` in it.
You can see that if you look at `add_summoner.py` there I also get the date from two endpoints and enrich the summoner and save him

**Comment by renja-g** [comment]
Also, I appreciate your help 👍

**Comment by shahuronghe** [comment]
Would you like to join me on google meet? shahu.mw@gmail.com

**Comment by renja-g** [comment]
Would Discord be fine?

**Comment by renja-g** [comment]
Fixed with #2
